### DISCUSSION TITLE
Anyone figured out how to display a WebRTC stream on the dashboard of Home Assistant?

### DISCUSSION THREAD

**Discussion by bobloadmire** [mrlt8/docker-wyze-bridge#666] [Q&A]
Love that the new update enables WebRTC streaming, but I don't know how to get the generic camera card to display the stream. It seems to be geared toward RTSP.

**Comment by tbrausch** [comment]
I have the same question.  I can display the stream in Home Assistant using rtsp://192.168.1.119:8554/spare to create the generic camera, but the stream rtsp://192.168.1.119:8889/spare is not recognized.  When running as a HA Addon, do we need to somehow define port 8889 or is this enabled by default (the documentation mentions that nothing special is required in Home Assistant Mode, which I assume means running as an Addon).  Also, do we need to configure the RTSPtoWebRTC integration in HA?

**Comment by bobloadmire** [comment]
rtsp works fine for me, but i'm trying to use WebRTC.

On Wed, Jan 11, 2023 at 1:56 PM tbrausch ***@***.***> wrote:

> I have the same question. I can display the stream in Home Assistant using
> rtsp://192.168.1.119:8554/spare to create the generic camera, but the
> stream rtsp://192.168.1.119:8889/spare is not recognized. When running as
> a HA Addon, do we need to somehow define port 8889 or is this enabled by
> default (the documentation mentions that nothing special is required in
> Home Assistant Mode, which I assume meand running as an Addon).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrlt8/docker-wyze-bridge/discussions/666#discussioncomment-4661258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAK3FBYULR3WKYCZ2EEKWTWR4T7ZANCNFSM6AAAAAATYPZSCU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
- Brad Stewart

**Reply by tbrausch** [reply]
I think what I said is that I am having the same issue, but I'm not sure if there is some additional setup required.  I also cannot display a WebRTC stream directly in a browser using http://192.168.1.119:8889/spare/ - this simply reports the error 
```
This site can’t be reached
192.168.1.119 refused to connect.
```
which I am assuming means that port 8889 is not being defined correctly

**Reply by mrlt8** [reply]
Can you access the WebRTC stream from the button on the WebUI `http://homeassistant.local:5000` or directly `http://homeassistant.local:8889/spare/`?

**Reply by tbrausch** [reply]
I'm not sure what you mean be accessing the WebRTC stream from the button on the WebUI.  There are only 3 buttons for the camera on the WebUI - HLS, RTMP, and RTSP (see below).  Also, as I stated above, trying to access the stream directly from http://homeassistant.local:8889/spare/ results in the error
```
This site can’t be reached
192.168.1.119 refused to connect.
```
I'm sure I am missing something obvious, but cannot figure out what it is.  Just to reiterate, I am running Wyze Bridge from HA as an addon.

![spare](https://user-images.githubusercontent.com/45288916/211951324-ba19f31a-caf3-4b37-9f64-3c4228fb14d1.jpg)

**Reply by mrlt8** [reply]
hmm, that is strange, there should be four buttons. You should be seeing an extra "WebRTC" button before HLS button.

Do you see the new ports in the config page for the bridge? 
![ports](https://user-images.githubusercontent.com/67088095/211958939-69984efa-af1b-44ff-9854-830e64bf3f91.PNG)

Additionally, you _may_ need to set the bridge IP if it can't find it automatically:
![bridge ip](https://user-images.githubusercontent.com/67088095/211959042-55049448-9ef4-4973-8c01-e71883a73a09.PNG)

**Reply by tbrausch** [reply]
I added the following in my config
```
WB_IP: 192.168.1.119
```
and now I see the WebRTC option in the Web UI, but when clicking it I get the error 

`192.168.1.119 refused to connect`

But I still do not see port 8889 in the Network Config

![network](https://user-images.githubusercontent.com/45288916/211960902-480ecf28-14cb-44a3-9ea6-50fdcbae6d50.jpg)

**Reply by mrlt8** [reply]
hmm, seems like it only did a partial upgrade to v1.11.0? Any chance you could reinstall the add-on to see if that gets the WebRTC ports to show up?

**Reply by tbrausch** [reply]
Thank you!  Reinstalling did the trick - also had to manually add the WB_IP parameter via YAML.  Displaying the camera with the Web UI in WebRTC mode or directly in the browser with

`http://192.168.1.119:8889/spare/`

now works perfectly.  However I still cannot get a generic camera defined and working with WebRTC correctly.  Is this even something that is supported?  If so, what should I use as a stream source?  **rtsp://192.168.1.119:8555/spare** works but has a slight delay, similar to MSE.  **rtsp://192.168.1.119:8889/spare** as well as **http://192.168.1.119:8889/spare/** fails to create the camera.

Update: Thank you so much for your help here, but it appears I am trying to do something that is not really supported in Wyze Bridge.

**Reply by mrlt8** [reply]
hmm will have to look into why `WB_IP` isn't getting set correctly (it should be getting the IP from supervisor/network/info on startup).

Do you know if any of the cards actively support WebRTC as a source?

**Reply by danielbublil** [reply]
> Thank you! Reinstalling did the trick - also had to manually add the WB_IP parameter via YAML. Displaying the camera with the Web UI in WebRTC mode or directly in the browser with
> 
> `http://192.168.1.119:8889/spare/`
> 
> now works perfectly. However I still cannot get a generic camera defined and working with WebRTC correctly. Is this even something that is supported? If so, what should I use as a stream source? **rtsp://192.168.1.119:8555/spare** works but has a slight delay, similar to MSE. **rtsp://192.168.1.119:8889/spare** as well as **http://192.168.1.119:8889/spare/** fails to create the camera.
> 
> Update: Thank you so much for your help here, but it appears I am trying to do something that is not really supported in Wyze Bridge.

where i need to set the WB_IP var?
i dont have in the addon configuration as @mrlt8 posted

found where to add it.
webrtc not working, all stream are choppy 
2023/01/12 23:21:07 [WyzeBridge] 172.30.32.2 - - [12/Jan/2023 23:21:07] "GET /snapshot/cam1.jpg?1673558463600= HTTP/1.1" 200 -
Traceback (most recent call last):
  File "/app/rtsp_event.py", line 5, in <module>
    import signal
  File "/usr/local/lib/python3.10/signal.py", line 7, in <module>
    _IntEnum._convert_(
  File "/usr/local/lib/python3.10/enum.py", line 563, in _convert_
    cls = cls(name, members, module=module)
  File "/usr/local/lib/python3.10/enum.py", line 387, in __call__
    return cls._create_(
  File "/usr/local/lib/python3.10/enum.py", line 518, in _create_
    enum_class = metacls.__new__(metacls, class_name, bases, classdict)
  File "/usr/local/lib/python3.10/enum.py", line 302, in __new__
    if canonical_member._value_ == enum_member._value_:
KeyboardInterrupt

**Comment by bobloadmire** [comment]
hmm, i'm forwarding through cloudflare and using HTTPS fine, so my setup is
a bit different than yours.

On Wed, Jan 11, 2023 at 2:30 PM tbrausch ***@***.***> wrote:

> I think what I said is that I am having the same issue, but I'm not sure
> if there is some additional setup required. I also cannot display a WebRTC
> stream directly in a browser using http://192.168.1.119:8889/spare/ -
> this simply reports the error
>
> This site can’t be reached
>
> 192.168.1.119 refused to connect.
>
>
> which I am assuming means that port 8889 is not being defined correctly
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrlt8/docker-wyze-bridge/discussions/666#discussioncomment-4661474>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAK3FF6YDJ6JN5YCNMJEGLWR4X77ANCNFSM6AAAAAATYPZSCU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
- Brad Stewart

**Comment by mrlt8** [comment]
You'd probably need to use something like `ws://homeassistant.local:8899/cam-name/ws`?

**Reply by bobloadmire** [reply]
I can view the cams fine with clourflaredurl.com/nameofcam

I want to have the HA dashboard generic camera card show the WebRTC stream since it seems to be a bit snappier than the RTSP stream.

**Comment by bobloadmire** [comment]
No I mean the dashboard. I want a tab on the dashboard that has the webrtc
streams. I guess I can use the web page card

**Reply by mrlt8** [reply]
Does the dashboard support WebRTC as a source? I looked into it in the past, but could never figure it out. Seems like it can do rtsp-to-WebRTC, but not WebRTC-to-WebRTC?
